htdocs/core/customreports.php

```diff
@@ -887,7 +926,26 @@
 	$sqlfilters = $search_component_params_hidden;
 	if ($sqlfilters) {
 		$sql .= forgeSQLFromUniversalSearchCriteria($sqlfilters, $errormessage, 0, 0, 1);
+
+		// Replace year, month, day with %Y, %Y-%m, %Y-%m-%d. For example, (t.datef-year := '2024') becomes WHERE DATE_FORMAT(t.datef, '%Y') = '2024'
```

Comment:
We must not use DATE_FORMAT to filter on date. This is not enable to use the index on date.
Instead you can replace all this block to use the function
`dolSqlDateFilter(...)`